### PR TITLE
✅ Improve test coverage

### PR DESCRIPTION
**Pull request by lig**
## Change Summary

Improve test coverage

## Related issue number

None

## Checklist

* [X] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @adriangb

### CODE REVIEW

**Comment by lig** [comment]
please review

File: pydantic/_internal/_decorators.py [hunk 1]
```diff
@@ -489,26 +489,17 @@ def unwrap_wrapped_function(
...
+    all: set[Any] = {partial, partialmethod, property}
+
     try:
         from functools import cached_property  # type: ignore
-
-        cached_property_list = [cached_property]
     except ImportError:
-        cached_property = int  # anything that doesn't match isinstance below
-        cached_property_list = []
+        cached_property = type('', (), {})  # type: ignore
```
Comment (samuelcolvin) [inline pydantic/_internal/_decorators.py:497]: Why is this required?

**Review by samuelcolvin — COMMENTED** [review]
Otherwise lgtm.

Who wrote this originally? They should probably review this.

**Comment by samuelcolvin** [comment]
Assigning  to @adriangb as he wrote the original I think.

File: pydantic/_internal/_decorators.py (hunk 1 above)
Comment (adriangb) [reply pydantic/_internal/_decorators.py:497]: creating a new type to reduce branching later I'm guessing?

**Review by adriangb — APPROVED** [review]

**Comment by samuelcolvin** [comment]
@adriangb is happy, so merged.
